packages/autometrics/src/wrappers.ts

```diff
@@ -478,7 +490,50 @@ type AutometricsDecoratorOptions<F> = F extends FunctionSig
  *
  * @group Wrapper and Decorator API
  */
-export function Autometrics<T extends Function | Object>(
+export function Autometrics<T extends FunctionSig>(
+  autometricsOptions: AutometricsOptions<T> = {},
+) {
+  return (target: T, context: DecoratorContext): T | void => {
+    switch (context.kind) {
+      case "class": {
+        const className =
+          typeof context.name === "string" ? context.name : target.name;
+        context.addInitializer(function () {
+          const classDecorator = getAutometricsClassDecorator({
+            className,
+            ...autometricsOptions,
+          });
+          classDecorator(this);
+        });
+      }
+
+      case "method": {
+        const functionName =
+          typeof context.name === "string" ? context.name : target.name;
+        return autometrics(
+          { functionName, ...autometricsOptions },
+          target,
+        ) as unknown as T;
+      }
+
+      default:
+        throw new Error(
```

Comment:
Feeling slightly uncomfortable about throwing from a library - maybe we could just log a console.err and pass through whatever was decorated unchanged?